### PR TITLE
Add service worker for better offline experience

### PR DESCRIPTION
**Pull request by ryansydnor**

### CODE REVIEW

File: sw-precache-config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  staticFileGlobs: [
+    'build/*.html',
+    'build/*.json',
+    'build/static/**/!(*map*)',
+    'ssr/**/*'
+  ],
+  runtimeCaching:[
+    {
+        urlPattern: /(bootstrap|fonts|polyfill|raven|btncdn|untappd|gstatic|photoreference)/,
```
Comment (ryansydnor) [inline sw-precache-config.js]: this is really aggressive. i think it means we'll have a pretty good offline experience at the risk of potentially stale stuff. i'm not 100% sure yet though

File: src/BreweryMap/BreweryMap.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const BreweryMap = ({
                key={ breweryKey }
                lat={ breweries[breweryKey].location.lat }
                lng={ breweries[breweryKey].location.lng }
-               pageUrl={ `${activeCity}/${breweryKey}` }
+               pageUrl={ `/${activeCity}/${breweryKey}` }
```
Comment (ryansydnor) [inline src/BreweryMap/BreweryMap.js]: whoops

File: src/index.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ ReactDOM.render(
                     replace(`/${nextState.params.city}`)
             }/>
         </Route>
+        <Redirect from="*" to="/nyc/" />
```
Comment (ryansydnor) [inline src/index.js]: we should make a cute 404 page

Comment (Aturberv) [inline src/index.js]: Defff

**Pull request merged**